[onnx][quantization] Add quantized concat conversion

supriyar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#30887 [onnx][quantization] Add quantized concat conversion**

Summary:
Support to convert quantized concat from pytorch to caffe2

Test Plan:
python test/onnx/test_pytorch_onnx_caffe2_quantized.py TestQuantizedOps.test_cat
Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D18855676](https://our.internmc.facebook.com/intern/diff/D18855676)

supriyar committed:
[onnx][quantization] Add quantized concat conversion
Summary:
Support to convert quantized concat from pytorch to caffe2

Test Plan:
python test/onnx/test_pytorch_onnx_caffe2_quantized.py TestQuantizedOps.test_cat
Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

houseroad reviewed (approved):
Looks good, ship it!

supriyar committed:
Update on "[onnx][quantization] Add quantized concat conversion"


Summary:
Support to convert quantized concat from pytorch to caffe2

Test Plan:
python test/onnx/test_pytorch_onnx_caffe2_quantized.py TestQuantizedOps.test_cat
Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D18855676](https://our.internmc.facebook.com/intern/diff/D18855676)

[ghstack-poisoned]

boydyang1 commented:
Hi,
I get the same error when I try to convert pytorch int8 maxpool to onnx.  
First, I run 
https://github.com/boydyang1/test_pytorch_int8_maxpool/blob/master/torch2onnx_maxpool.py
The error is "error loading ONNX model. Maximum call stack size exceeded in 'test_int8.onnx'."
What should I do?

boydyang1 commented:
Hi,
I get the same error when I try to convert pytorch int8 **maxpool** to onnx to caffe2.
1) First, I run torch2onnx_maxpool.py(the code list bellow) to get onnx and get this warning:
"symbolic_helper.py:176: UserWarning: ONNX export failed on max_pool2d because input size not accessible not supported"

2) when I run run_onnx_maxpool.py to load onnx model, I get this error 
"Exception: Don't know how to translate op max_pool2d"

What should I do?  The code is here
https://github.com/boydyang1/test_pytorch_int8_maxpool/blob/master/torch2onnx_maxpool.py
https://github.com/boydyang1/test_pytorch_int8_maxpool/blob/master/run_onnx_maxpool.py
@supriyar @houseroad @suo
